ffmpeg.c

```diff
@@ -891,11 +964,16 @@ static int ffmpeg_flush_codec(struct ffmpeg *ffmpeg){
                     my_packet_unref(ffmpeg->pkt);
                     return -1;
                 }
-                retcd = av_write_frame(ffmpeg->oc, &ffmpeg->pkt);
-                if (retcd < 0) {
-                    MOTION_LOG(ERR, TYPE_ENCODER, NO_ERRNO
-                        ,_("Error writing draining video frame"));
-                    return -1;
+                // v4l2_m2m encoder uses pts 0 and size 0 to indicate AVERROR_EOF
+                if ((ffmpeg->pkt.pts > 0) && (ffmpeg->pkt.size > 0)) {
+                    retcd = av_write_frame(ffmpeg->oc, &ffmpeg->pkt);
+                    if (retcd < 0) {
+                        MOTION_LOG(ERR, TYPE_ENCODER, NO_ERRNO
+                            ,_("Error writing draining video frame"));
+                        return -1;
+                    }
+                } else {
+                    recv_cd = AVERROR_EOF;
```

Comment:
This will work but it is a bit inconsistent with the other sections.  They are logging the message and returning -1.

Reply:
Is the code easier to read if I replace it with this block:
```c
                // v4l2_m2m encoder uses pts 0 and size 0 to indicate AVERROR_EOF
                if ((ffmpeg->pkt.pts == 0) || (ffmpeg->pkt.size == 0)) {
                    recv_cd = AVERROR_EOF;
                    my_packet_unref(ffmpeg->pkt);
                    continue;
                }
                retcd = av_write_frame(ffmpeg->oc, &ffmpeg->pkt);
                if (retcd < 0) {
                    MOTION_LOG(ERR, TYPE_ENCODER, NO_ERRNO
                        ,_("Error writing draining video frame"));
                    return -1;
                }
```

Basically the `av_write_frame` chunk remains the same as before. The only new bit is to look for PTS or size zero, then change the return code to `AVERROR_EOF` to fall out of the loop. I could even replace the `recv_cd = AVERROR_EOF` and `continue` with a single break, but some people hate breaking loops as coding convention.